metricflow/test/snapshots/test_dataflow_to_sql_plan.py/SqlQueryPlan/DuckDB/test_derived_metric_with_offset_to_grain_and_granularity__plan0_optimized.sql

```diff
@@ -0,0 +1,63 @@
+-- Compute Metrics via Expressions
+SELECT
+  metric_time__week
+  , bookings - bookings_at_start_of_month AS bookings_growth_since_start_of_month
+FROM (
+  -- Combine Metrics
+  SELECT
+    COALESCE(subq_18.metric_time__week, subq_26.metric_time__week) AS metric_time__week
+    , subq_18.bookings AS bookings
+    , subq_26.bookings_at_start_of_month AS bookings_at_start_of_month
+  FROM (
+    -- Aggregate Measures
+    -- Compute Metrics via Expressions
+    SELECT
+      metric_time__week
+      , SUM(bookings) AS bookings
+    FROM (
+      -- Read Elements From Semantic Model 'bookings_source'
+      -- Metric Time Dimension 'ds'
+      -- Pass Only Elements:
+      --   ['bookings', 'metric_time__week']
+      SELECT
+        DATE_TRUNC('week', ds) AS metric_time__week
+        , 1 AS bookings
+      FROM ***************************.fct_bookings bookings_source_src_10001
+    ) subq_16
+    GROUP BY
+      metric_time__week
+  ) subq_18
+  INNER JOIN (
+    -- Join to Time Spine Dataset
+    -- Pass Only Elements:
+    --   ['bookings', 'metric_time__week']
+    -- Aggregate Measures
+    -- Compute Metrics via Expressions
+    SELECT
+      DATE_TRUNC('week', subq_22.ds) AS metric_time__week
+      , SUM(subq_20.bookings) AS bookings_at_start_of_month
+    FROM ***************************.mf_time_spine subq_22
+    INNER JOIN (
+      -- Read Elements From Semantic Model 'bookings_source'
+      -- Metric Time Dimension 'ds'
+      SELECT
+        ds AS metric_time__day
+        , 1 AS bookings
+      FROM ***************************.fct_bookings bookings_source_src_10001
+    ) subq_20
+    ON
+      DATE_TRUNC('month', subq_22.ds) = subq_20.metric_time__day
+    GROUP BY
+      DATE_TRUNC('week', subq_22.ds)
```

Comment:
I think something's wrong here. Is this supposed to show bookings at start of month for every week in the time range?

Right now, it looks like it'll show bookings at start of month for weeks that include the first day of the month, because the INNER JOIN filters out a bunch of dates. Then there's no matching value in the inner join expression below for any week that's missing from this subset.